### PR TITLE
Improve YlmSpherepack documentation

### PR DESCRIPTION
**Pull request by nikwit**
## Proposed changes
The current pieces of documentation in `YlmSpherepack` are impossible to understand for someone unfamiliar with SPHEREPACK. I added some general documentation, explaining how the spherical expansions are represented.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Comment by markscheel** [comment]
Failing test is that docs don't render properly.

**Comment by nikwit** [comment]
sorry for that, I had fixed a typo with another typo.

File: src/ApparentHorizons/YlmSpherepack.hpp [hunk 1]
```diff
@@ -16,8 +16,59 @@
...
+ * \details The class `YlmSpherepack` defines the C++ interface to the fortran
+ * library SPHEREPACK used for computations on the surface of a sphere.
+ *
+ * Given a real-valued, scalar function \f$g(\theta, \phi)\f$, SPHEREPACK
+ * expands it as:
+ *
+ * \f{align*}
+ * g(\theta, \phi)
+ * &=\frac{1}{2}\sum_{l=0}^{l_{max}}\bar P_l^0(\cos\theta) a_{l0}
+ * +\sum_{l=1}^{l_{max}}\sum_{m=1}^{min(l, m_{max})}\bar P_l^m(\cos\theta)\{
```
Comment (nilsdeppe) [inline src/ApparentHorizons/YlmSpherepack.hpp]: `min` -> `\min` or `\mathrm{min}` (same with `max`)

File: src/ApparentHorizons/YlmSpherepack.hpp [hunk 2]
```diff
@@ -16,8 +16,59 @@
...
+ * the collocation points can be computed wth the method `theta_phi_points`.
+ *
+ * To convert between the two representations the methods `spec_to_phys` and
+ * `phys_to_spec` can be used. For internal calculations SPHEREPACK will usually
+ * convert to spectral coefficients first so it is in general more efficient to
+ * use these directly.
+ *
+ * Most methods of SPHEREPACK will compute the requested values of e.g.
+ * `gradient` or `scalar_laplacian` at the collocation points, effectively
+ * returning an expansion of form 2. To evaluate the function at arbitrary
```
Comment (nilsdeppe) [inline src/ApparentHorizons/YlmSpherepack.hpp]: sorry, what do you mean by `expansion of form 2`?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
please squash directly

File: src/ApparentHorizons/YlmSpherepack.hpp (hunk 2 above)
Comment (nikwit) [reply src/ApparentHorizons/YlmSpherepack.hpp]: I mean the second representation of SPHEREPACK as I defined it in the enumeration above. I made it a bit clearer.

File: src/ApparentHorizons/YlmSpherepack.hpp [hunk 3]
```diff
@@ -16,8 +16,60 @@
...
+ * the collocation points can be computed wth the method `theta_phi_points`.
+ *
+ * To convert between the two representations the methods `spec_to_phys` and
+ * `phys_to_spec` can be used. For internal calculations SPHEREPACK will usually
+ * convert to spectral coefficients first so it is in general more efficient to
+ * use these directly.
+ *
+ * Most methods of SPHEREPACK will compute the requested values of e.g.
+ * `gradient` or `scalar_laplacian` at the collocation points, effectively
+ * returning an expansion in the second representation of SPHEREPACK. To
```
Comment (nilsdeppe) [inline src/ApparentHorizons/YlmSpherepack.hpp]: sorry, I have no idea from the documentation here that there are two representations. Are you talking about modal vs nodal? If so, I would use that wording

**Comment by nikwit** [comment]
I changed it to modal and nodal form now.

**Review by nilsdeppe — DISMISSED** [review]

File: src/ApparentHorizons/YlmSpherepack.hpp [hunk 4]
```diff
@@ -16,8 +16,60 @@
...
+ *
+ * and \f$P_{nm}(x)\f$ are the associated Legendre polynomials.
+ *
+ * Internally, SPHEREPACK can represent such an expansion in two ways which we
+ * will refer to as modal and nodal representations:
+ *
+ * -# modal: The spectral coefficient arrays \f$a_{lm}\f$ and \f$b_{lm}\f$,
+ * referred to as `spectral_coefs` in the methods below. For this C++ interface,
+ * they are saved in a single `DataVector`. To help you index the coefficients
+ * as expected by this interface use the class `SpherepackIterator`.
```
Comment (markscheel) [inline src/ApparentHorizons/YlmSpherepack.hpp]: comma after 'interface'

File: src/ApparentHorizons/YlmSpherepack.hpp [hunk 5]
```diff
@@ -16,8 +16,60 @@
...
+ *
+ * -# nodal: The values at certain collocation points, referred to as
+ * `collocation_values` in the methods below. This is an array of the expanded
+ * function evaluated at collocation angles \f$g(\theta_i, \phi_j)\f$, where
```
Comment (markscheel) [inline src/ApparentHorizons/YlmSpherepack.hpp]: Wording sounds like the collocation angles are called 'g'.  How about
`This is an array of the expanded function \f$g(\theta,\phi)\f$ evaluated at collocation values \f$(\theta_i,\phi_j)\f$, where...`  with no 'g' in the 2nd formula.

File: src/ApparentHorizons/YlmSpherepack.hpp [hunk 6]
```diff
@@ -16,8 +16,60 @@
...
+ * \f$\theta_i\f$ are Gauss-Legendre quadrature nodes in the interval \f$(0,
+ * \pi)\f$ with \f$i = 0, ..., l_{\max}\f$, and \f$\phi_j\f$ is distributed
+ * uniformly in \f$(0, 2\pi)\f$ with \f$i = 0, ..., 2m_{\max}\f$. The angles of
+ * the collocation points can be computed wth the method `theta_phi_points`.
+ *
+ * To convert between the two representations the methods `spec_to_phys` and
+ * `phys_to_spec` can be used. For internal calculations SPHEREPACK will usually
+ * convert to spectral coefficients first so it is in general more efficient to
```
Comment (markscheel) [inline src/ApparentHorizons/YlmSpherepack.hpp]: comma after 'first'

File: src/ApparentHorizons/YlmSpherepack.hpp [hunk 7]
```diff
@@ -16,8 +16,60 @@
...
+ * use these directly.
+ *
+ * Most methods of SPHEREPACK will compute the requested values of e.g.
+ * `gradient` or `scalar_laplacian` at the collocation points, effectively
+ * returning an expansion in nodal form as defined above. To evaluate the
+ * function at arbitrary angles \f$\theta\f$, \f$\phi\f$ these values have to be
```
Comment (markscheel) [inline src/ApparentHorizons/YlmSpherepack.hpp]: comma after phi

File: src/ApparentHorizons/YlmSpherepack.hpp [hunk 8]
```diff
@@ -16,8 +16,60 @@
...
+ * they are saved in a single `DataVector`. To help you index the coefficients
+ * as expected by this interface use the class `SpherepackIterator`.
+ *
+ * -# nodal: The values at certain collocation points, referred to as
+ * `collocation_values` in the methods below. This is an array of the expanded
+ * function evaluated at collocation angles \f$g(\theta_i, \phi_j)\f$, where
+ * \f$\theta_i\f$ are Gauss-Legendre quadrature nodes in the interval \f$(0,
+ * \pi)\f$ with \f$i = 0, ..., l_{\max}\f$, and \f$\phi_j\f$ is distributed
+ * uniformly in \f$(0, 2\pi)\f$ with \f$i = 0, ..., 2m_{\max}\f$. The angles of
+ * the collocation points can be computed wth the method `theta_phi_points`.
```
Comment (markscheel) [inline src/ApparentHorizons/YlmSpherepack.hpp]: wth -> with

**Review by markscheel — CHANGES_REQUESTED** [review]
Just minor changes. Please squash directly.

**Comment by nikwit** [comment]
squashed, thanks for your reviews :)

**Review by nilsdeppe — APPROVED** [review]

**Review by markscheel — APPROVED** [review]

**Pull request merged**